### PR TITLE
Feature/add avatar to user

### PR DESCRIPTION
**Pull request by AlanDavd**
Add avatar field to user model

### CODE REVIEW

File: src/modules/user/user.entity.ts [hunk 1]
```diff
@@ -21,6 +21,8 @@ export class User extends BaseEntity {
   password: string;
   @Column({ name: 'is_active', default: false })
   isActive: boolean;
+  @Column({ type: 'varchar', name: 'avatar', length: 255 })
+  avatar: string;
```
Comment (jcmexdev) [inline src/modules/user/user.entity.ts:25]: 😎 good, can you add this field to data model please

File: src/modules/database/migrations/1606322010210-create_menu_dishes_table.ts [hunk 2]
```diff
@@ -1,18 +1,28 @@
...
+  public async down(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `COMMENT ON COLUMN "menu_categories"."author_id" IS NULL`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "menu_categories" ALTER COLUMN "author_id" DROP NOT NULL`,
+    );
+    await queryRunner.query(`DROP TABLE "menu-dishes"`);
+  }
 }
```
Comment (jcmexdev) [inline src/modules/database/migrations/1606322010210-create_menu_dishes_table.ts:28]: why this file has changes?, it doesn't have nothing related with your feature

File: src/modules/user/user.entity.ts (hunk 1 above)
Comment (AlanDavd) [reply src/modules/user/user.entity.ts:25]: Which data model?
It is already in the entity and dto's

File: src/modules/user/user.service.ts [hunk 3]
```diff
@@ -41,4 +46,25 @@ export class UserService {
...
+    const user = await this._userRepository.findOne({ id: userId });
+    if (user) throw new NotFoundException('user_not_found');
+    if (user.id != userId) throw new ConflictException('user_not_match');
+    user.firstname = newUser.firstname;
+    user.lastname = newUser.lastname;
+    return plainToClass(UserResponseDto, await user.save());
+  }
+
+  async uploadAvatar(userId: number, avatarUrl: string) {
+    const user = await this._userRepository.findOne({ id: userId });
```
Comment (jlazo) [inline src/modules/user/user.service.ts:66]: What happen if the user is undefined?

File: src/modules/user/user.controller.ts [hunk 4]
```diff
@@ -20,22 +37,81 @@ export class UserController {
...
+  @ApiNotFoundResponse({
+    description: 'It happens when user not found (user_not_found)',
+  })
+  disable(@Param('id') userId: number) {
+    return this._userService.disable(userId);
+  }
+
+  @Get('avatar/:fileId')
+  @UseGuards(AuthGuard('jwt'))
+  async serveDishImage(@Param('fileId') fileId, @Res() res): Promise<any> {
```
Comment (jlazo) [inline src/modules/user/user.controller.ts]: Why the method name is serveDishImage?

Comment (AlanDavd) [reply src/modules/user/user.controller.ts]: Fixed

File: src/modules/user/user.service.ts (hunk 3 above)
Comment (AlanDavd) [reply src/modules/user/user.service.ts:66]: I used exceptions to handle undefined user but I've always got error when uploading an image even though the image is saved. Without the next exception, the api works fine, with them I always got error even when everything worked well
if (user) throw new NotFoundException('user_not_found');
if (user.id != userId) throw new ConflictException('user_not_match');

**Review by jlazo — APPROVED** [review]

File: src/modules/user/user.service.ts [hunk 5]
```diff
@@ -41,4 +46,25 @@ export class UserService {
     theUser.password = await hash(password, salt);
     return plainToClass(UserResponseDto, await theUser.save());
   }
+
+  async disable(userId: number) {
+    const user = await this._userRepository.findOne({ id: userId });
+    if (user) throw new NotFoundException('user_not_found');
```
Comment (jcmexdev) [inline src/modules/user/user.service.ts:52]: it should be a negation right? if (!user)

File: src/modules/user/user.service.ts [hunk 6]
```diff
@@ -41,4 +46,25 @@ export class UserService {
...
+    user.isActive = false;
+    return user.save();
+  }
+  async update(userId: number, newUser: UserUpdateDto) {
+    const user = await this._userRepository.findOne({ id: userId });
+    if (user) throw new NotFoundException('user_not_found');
```
Comment (jcmexdev) [inline src/modules/user/user.service.ts:58]: same negative here !user

**Review by jcmexdev — APPROVED** [review]

**Pull request merged**